File: UI/src/app/shared/models/job-history.ts
import { City } from "./city";
import { Country } from "./country";
import { DocumentList } from "./documentlist";
import { employee } from "./employee";

import { EmployeeType } from "./employeeType";
import { State } from "./state";

export interface JobHistory {
    id: string;
    companyName: string;
    positionHeld: string;
    employmentType: string;
    employmentTypeName:string;
    stateId:string;
    stateName:string;
    zipCode: number;
    city: string;
    cityName:string;
    country: string;
    countryName:string; 
    startDate: string; 
    endDate: string; 
    document: string;
    leavingReason: string;
    employeeId: string;
    employee?: employee | null;
    cityDetails?: City | null;
    countryDetails?: Country | null;
    employeeTypeDetails?:EmployeeType|null;
    action: number;
    filename?:any;
  }
  

File: API/beONHR.Entities/JobHistory.cs
using beONHR.Entities.User;
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace beONHR.Entities
{
    public class JobHistory
    {
        [Key]
        public Guid Id { get; set; }
        public string CompanyName {  get; set; }
        public string PositionHeld {  get; set; }
        public Guid EmploymentType { get; set; }
        public int ZipCode { get; set; }
        public Guid City { get; set;} //forignkey
        public Guid Country { get; set;} //forignkey
        public Guid StateId { get; set;} //forignkey
        public DateOnly StartDate{ get; set;} 
        public DateOnly EndDate{ get; set;} 
        public string Document {  get; set;}
        public string LeavingReason { get; set;}
        public Guid EmployeeId { get; set; } //forignkey
        public bool IsDeleted { get; set; } = false;

        [ForeignKey("EmploymentType")]
        public virtual TypeofEmployment? EmployeeType { get; set; }

        [ForeignKey("EmployeeId")]
        public virtual Employee? Employee { get; set; }

        [ForeignKey("City")]
        public virtual City? cityId { get; set; }

        [ForeignKey("Country")]
        public virtual Country? countryId{ get; set; }


        [ForeignKey("StateId")]
        public virtual State? State { get; set; }

    }
}
